src/org/zaproxy/zap/extension/ascanrules/TestSQLInjection.java

```diff
@@ -1400,28 +1400,23 @@ private void expressionBasedAttack(String param, String modifiedParamValue, Stri
 				} catch (SocketException ex) {
 					if (log.isDebugEnabled()) log.debug("Caught " + ex.getClass().getName() + " " + ex.getMessage() +
 							" when accessing: " + msgConfirm.getRequestHeader().getURI().toString());
-					continue; //Something went wrong, continue to the next item in the loop
+					return; //Something went wrong
 				}
 				countExpressionBasedRequests++;
 
 				String confirmExpressionOutputUnstripped = msgConfirm.getResponseBody().toString();
-				String confirmExpressionOutputStripped = this.stripOff(confirmExpressionOutputUnstripped, modifiedParamValueConfirm);
+				String confirmExpressionOutputStripped = stripOffOriginalAndAttackParam(confirmExpressionOutputUnstripped, originalParam, modifiedParamValueConfirm);
 
-				//set up two little arrays to ease the work of checking the unstripped output or the stripped output
-				String confirmExpressionBodyOutput[] = {confirmExpressionOutputUnstripped, confirmExpressionOutputStripped};
+				//give variable a nicer name
+				String confirmExpressionBodyOutput = confirmExpressionOutputStripped;
```

Comment:
If the variable name isn't useful for development/maintenance use then just replace it throughout. No point wasting memory storing the same string twice or CPU cycles assigning it twice.